langserver/handler.go

```diff
@@ -73,7 +73,14 @@ type LangHandler struct {
 
 	cancel *cancel
 
-	Config Config // language handler configuration; must not change after handling has begun
+	// DefaultConfig is the default values used for configuration. It is
+	// combined with InitializationOptions after initialize. This should be
+	// set by LangHandler creators. Please read config instead.
+	DefaultConfig Config
+
+	// config is the language handler configuration. It is Default with
+	// InitializationOptions applied.
+	config *Config // pointer so we panic if someone reads before we set it.
```

Comment:
config is the language handler configuration. It is a combination of `DefaultConfig` and `InitializationOptions`.